pilot/pkg/model/service.go

```diff
@@ -527,6 +527,34 @@ func (h Hostname) Matches(o Hostname) bool {
 	return matches
 }
 
+func (h Hostname) SubsetOf(o Hostname) bool {
+	if len(h) == 0 && len(o) == 0 {
+		return true
+	}
+
+	hWildcard := string(h[0]) == "*"
+	oWildcard := string(o[0]) == "*"
+	if !oWildcard {
+		if hWildcard {
+			return false
+		}
+		return h == o
+	}
+
+	longer, shorter := string(h), string(o)
+	if hWildcard {
```

Comment:
shouldn't this be shorter, longer? Assuming this function does h.SubsetOf(o), implying h is a subset of o

Reply:
Ha ha, logically you'd think that way, but a subset will be either the same or longer than the other, e.g., `*.example.com` is a subset of `*.com`